### PR TITLE
Fix all warnings

### PR DESCRIPTION
**Pull request by crudelios**
All compiler warnings removed again (hopefully no new ones appear due to this PR 😉)

### CODE REVIEW

File: src/core/config.c [hunk 1]
```diff
@@ -73,7 +73,7 @@ void config_set_defaults(void)
     for (int i = 0; i < CONFIG_MAX_ENTRIES; ++i) {
         values[i] = default_values[i];
     }
-    strncpy(string_values[CONFIG_STRING_UI_LANGUAGE_DIR], default_string_values[CONFIG_STRING_UI_LANGUAGE_DIR], CONFIG_STRING_VALUE_MAX - 1);
+    strncpy(string_values[CONFIG_STRING_UI_LANGUAGE_DIR], default_string_values[CONFIG_STRING_UI_LANGUAGE_DIR], CONFIG_STRING_VALUE_MAX);
```
Comment (bvschaik) [inline src/core/config.c:76]: What was the warning here? The `- 1` is on purpose: if the string is longer than `CONFIG_STRING_VALUE_MAX`, the string is copied without trailing `\0`. Using `-1` leaves the last position unwritten so its default is `\0`, and the string will always be terminated.

Comment (crudelios) [reply src/core/config.c:76]: This is the warning:

```
warning: 'strncpy' output may be truncated copying 63 bytes from a string of length 63 [-Wstringop-truncation]
   76 |     strncpy(string_values[CONFIG_STRING_UI_LANGUAGE_DIR], default_string_values[CONFIG_STRING_UI_LANGUAGE_DIR], CONFIG_STRING_VALUE_MAX - 1);
```

One alternative is to use `snprintf`. It's slower but it copies the ending `0` when the size doesn't fit.

Comment (bvschaik) [reply src/core/config.c]: Ok. In this case (known src and dst), we can indeed use `strncpy` with `CONFIG_STRING_VALUE_MAX`.

Comment (crudelios) [reply src/core/config.c]: I went ahead and replaced with `snprintf` anyway. It's safer as it adds the `0` and the code is not performance critical.

Comment (bvschaik) [reply src/core/config.c]: I don't like that on a conceptual level: you don't use snprintf for copying a string.

Comment (crudelios) [reply src/core/config.c]: All right, I'll revert it 👍 

But although copying strings isn't the real purpose of `snprintf`, the truth is it works, it's safer than `strncpy` and it's actually used for that effect.

Comment (bvschaik) [reply src/core/config.c:76]: On second thought, `memcpy` might have been even more appropriate here, but let's stick with strncpy now ;)

**Pull request merged**